### DISCUSSION TITLE
Autoload external file without restarting container

### DISCUSSION THREAD

**Discussion by skoeswanto** [coredns/coredns#4561] [General]
I am using coredns/coredns docker image with a very simple setup as following:

Corefile
```
.:53 {
    forward . 8.8.8.8 9.9.9.9 1.1.1.1
    log
    errors
}

mypublicdomain.com:53 {
    file /root/mypublicdomain.com.db
    reload
    log
    errors
}
```

Inside mypublicdomain.com.db file:
```
mypublicdomain.com        IN  SOA  ns1.mypublicdomain.com. yudi.mypublicdomain.com 2018040502 7200 3600
1209600 3600

ns1      IN  A    real-public-ip
ns2      IN  A    real-public-ip
mail     IN  A    real-public-ip

subdomain1      IN  A    real-public-ip
subdomain2      IN  A    real-public-ip
subdomain3    IN  A    real-public-ip
```

Note after SOA ns1... I put my email yudi@mypublicdomain.com but from tutorial I read it should be in '.' (dot) format.

What I want is sometimes I add another subdomain into mypublicdomain.com.db file but despite waiting for hours the new entry is not showing up in nslookup.  I have to restart the docker container to get the new entry resolved.

Any help would be appreciated, I am very new in DNS hence might have some terms wrong.

I also tried:
```
mypublicdomain.com:53 {
    file /root/mypublicdomain.com.db {
        reload 30s
    }
    log
    errors
}
```
but the container just keep restarting with errors, so I reverted back putting reload outside the file plugin and inside mypublicdomain.com:53 zone.

Yudi

**Comment by chrisohaver** [comment]
The SOA serial must be updated for *file* to load the changes.

**Reply by skoeswanto** [reply]
Thanks much, I will try it.
